ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs

```diff
@@ -124,14 +134,14 @@ type EraCrypto era = Crypto era
 -- https://github.com/input-output-hk/ouroboros-network/issues/2890
 class ( SL.ShelleyBasedEra era
 
-        -- Constraints that relate to the protocol. These should be dropped once
-        -- the protocol is independent of the ledger.
-      , SL.PraosCrypto (EraCrypto era)
+        -- TODO This constraint is quite tight, since it fixes things to the
```

Comment:
To which constraint are we referring here? The one above or below? Or the one that was removed?

Reply:
Below. I always add comments above...